### PR TITLE
Search: Add tracks events

### LINKED ISSUE #8571: Search: Add Tracks events to customizer.
We've discussed this a number of times, but we don't seem to have an open issue and I don't think it's been done yet.

### PR DESCRIPTION
**Pull request by ebinnion**
Fixes #8571

This PR adds analytics to the Jetpack Search feature. At the moment, specifically in the widget UI.

I'd still like to figure out a way to track when a widget is added or updated that way we can understand how many widgets are being added to each site, which filters are most used, etc.

To test:

- Checkout branch on a site with a Jetpack Professional plan
- Add a Jetpack Search widget
- Make changes to widget
- Delete widget
- Check and ensure events were logged in tracks

### CODE REVIEW

File: modules/search/class.jetpack-search-widget-filters.php [hunk 1]
```diff
@@ -285,6 +312,8 @@ private function sorting_to_wp_query_param( $sort ) {
 	function update( $new_instance, $old_instance ) {
 		$instance = array();
 
+		error_log( print_r( array( $_GET, $_POST ), true ) );
```
Comment (zinigor) [inline modules/search/class.jetpack-search-widget-filters.php]: I don't think you meant to leave it here :)

**Comment by zinigor** [comment]
Also, I'm not sure if it's supposed to be this way, but for every event I see both `t.gif` and `g.gif` being loaded, is that ok?

File: modules/search/class.jetpack-search-widget-filters.php (hunk 1 above)
Comment (gravityrail) [reply modules/search/class.jetpack-search-widget-filters.php]: Fixed

**Comment by gravityrail** [comment]
@zinigor as per comment in the code, this fires both tracks and MC events, therefore it is intentional to load both t.gif and g.gif.

> This function will fire both a Tracks and MC stat

**Comment by ebinnion** [comment]
Thanks for pointing those out @zinigor! As a note, I'm not quite done. I plan on finishing this up today.

**Comment by ebinnion** [comment]
We also send events when a widget is created, updated, or deleted via PHP. So, if we find the UI events noisy, we could filter those out.

**Pull request merged**